AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLiteStorageEngineAdapterTests.swift

```diff
@@ -375,4 +375,120 @@ class SQLiteStorageEngineAdapterTests: BaseDataStoreTests {
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameAndFileExists() {
+        let fileExistsInvoked = expectation(description: "Version remains the same")
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionInUserDefaultsRemainsSame") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "previousVersion"
+
+        let mockFileManager = MockFileManager()
+        mockFileManager.fileExists = true
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        fileExistsInvoked.fulfill()
+
+        wait(for: [fileExistsInvoked], timeout: 10)
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameButFileDoesNotExist() {
+        let fileNotExistsInvoked = expectation(description: "Version remains the same")
+
+        guard let userDefaults = UserDefaults.init(suiteName: "VersionRemainsSameButFileDoesNotExist") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+
+        userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "previousVersion"
+
+        let mockFileManager = MockFileManager()
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        fileNotExistsInvoked.fulfill()
+
+        wait(for: [fileNotExistsInvoked], timeout: 10)
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsUpdateFail() {
```

Comment:
`testClearIfNewVersionFailure`
- previous != new so you can get pass the first check
- mock fileExist = true
- mock hasError = true
- catch and assert the error that you expected to throw, what type of DataStore error? fulfill not needed

Reply:
I changed to using this `XCTAssertEqual(error as? MockFileManagerError, MockFileManagerError.removeItemError)` inside catch block

Reply:
I kind of think it's bit weird to have 
`
userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)
let newVersion = "newVersion"
`
because `try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
                                                             dbFilePath: URL(string: "dbFilePath")!,
                                                             userDefaults: userDefaults,
                                                             fileManager: mockFileManager)` is not setting newVersion to userDefaults

Reply:
can u elaborate on what you mean? the method isn't setting `newVersion` to userDefaults,  it's just clearing if required, or clearing if version is different from what is stored. are u thinking the method should be named differently? 

in terms of a test, it just means that we expect database to be deleted when version differs, and if for some reason it was terminated and restarted then, version still differs (because newVersion is never set) then it will do no-op since file does not exist.

Reply:
Oh, sorry. I didn't think into this test very clearly when I replied. There is no problem now. 

But in terms of the first bullet in this conversation `previous != new so you can get pass the first check`
what do you mean by this? Could you explain a bit on it?

Reply:
i just meant to arrange the test so that the version in user defaults is different from the version passed in, you are already doing this